### ISSUE TITLE
Wraparound when exceeding history (max_file_size) does not work as expected

### ISSUE THREAD

**Issue by zwieberl** [MovingtoMars/liner#76]
Maybe this is meant this way, but if the history-file has exactly `max_file_size` lines and another is pushed, the file is cleared completely and the newly pushed is its only entry.
The in memory-history works as expected (drops the oldest, but keeps the rest).

How to reproduce (quick and dirty):
```bash
$ touch test
$ for x in $(seq 1000); do echo $x >> test; done
$ cat test
```
This gives 1000 entries.

Compile and run:
```rust
extern crate liner;
use liner::Context;

fn main() {
    let mut con = Context::new();
    con.history.set_file_name(Some("test".to_string()));
    con.history.load_history();
    let command = "blabla";
    con.history.push(command.into());
    con.history.commit_history();
}
```
and then
```bash
$ cat test
```
again. Result only "blabla" will be in the history-file.

**Comment by zwieberl** [comment]
```diff
diff --git a/src/history.rs b/src/history.rs
index 8545959..c5a23be 100644
--- a/src/history.rs
+++ b/src/history.rs
@@ -242,7 +242,7 @@ fn write_to_disk(max_file_size: usize, new_item: &Buffer, file_name: &str) -> io
                             total_read += 1;
                             if byte == b'\n' {
                                 stored += 1;
-                                if stored == max_file_size {
+                                if stored == (stored + cmds_read - max_file_size) {
                                     seek_point = total_read;
                                     break
                                 }
```

**Comment by MovingtoMars** [comment]
That patch does fix the functionality when using liner, but for some reason it adds a lot of nulls to the start of the history file.

I'll have a better look tomorrow.

**Comment by MovingtoMars** [comment]
This should be fixed as of f5e00be2d6d7f318871458060de293063136807f
